backends-common/elasticsearch/src/test/java/org/apache/james/backends/es/DockerElasticSearch.java

```diff
@@ -55,85 +76,188 @@ static ElasticSearchAPI from(Host esHttpHost) {
 
-    private static final int ES_HTTP_PORT = 9200;
+    interface Fixture {
+        int ES_HTTP_PORT = 9200;
+    }
 
-    private final DockerContainer eSContainer;
+    class NoAuth implements DockerElasticSearch {
 
-    public DockerElasticSearch() {
-        this(Images.ELASTICSEARCH_6);
-    }
+        private final DockerContainer eSContainer;
 
-    public DockerElasticSearch(String imageName) {
-        this.eSContainer = DockerContainer.fromName(imageName)
-            .withTmpFs(ImmutableMap.of("/usr/share/elasticsearch/data", "rw,size=200m"))
-            .withExposedPorts(ES_HTTP_PORT)
-            .withEnv("discovery.type", "single-node")
-            .withAffinityToContainer()
-            .waitingFor(new HostPortWaitStrategy().withRateLimiter(RateLimiters.TWENTIES_PER_SECOND));
-    }
+        public NoAuth() {
+            this(Images.ELASTICSEARCH_6);
+        }
 
-    public void start() {
-        if (!eSContainer.isRunning()) {
-            eSContainer.start();
+        public NoAuth(String imageName) {
+            this.eSContainer = DockerContainer.fromName(imageName)
+                .withTmpFs(ImmutableMap.of("/usr/share/elasticsearch/data", "rw,size=200m"))
+                .withExposedPorts(ES_HTTP_PORT)
+                .withEnv("discovery.type", "single-node")
+                .withAffinityToContainer()
+                .waitingFor(new HostPortWaitStrategy().withRateLimiter(RateLimiters.TWENTIES_PER_SECOND));
         }
-    }
 
-    public void stop() {
-        eSContainer.stop();
-    }
+        public void start() {
+            if (!eSContainer.isRunning()) {
+                eSContainer.start();
+            }
+        }
 
-    public int getHttpPort() {
-        return eSContainer.getMappedPort(ES_HTTP_PORT);
-    }
+        public void stop() {
+            eSContainer.stop();
+        }
+
+        public int getHttpPort() {
+            return eSContainer.getMappedPort(ES_HTTP_PORT);
+        }
+
+        public String getIp() {
+            return eSContainer.getHostIp();
+        }
+
+        public Host getHttpHost() {
+            return Host.from(getIp(), getHttpPort());
+        }
+
+        public void pause() {
+            eSContainer.pause();
+        }
 
-    public String getIp() {
-        return eSContainer.getHostIp();
+        public void unpause() {
+            eSContainer.unpause();
+        }
     }
 
-    public Host getHttpHost() {
-        return Host.from(getIp(), getHttpPort());
+    class WithAuth implements DockerElasticSearch {
+
+        private static final String DEFAULT_USERNAME = "elasticsearch";
+        private static final String DEFAULT_PASSWORD = "secret";
+        private static final ElasticSearchConfiguration.Credential DEFAULT_CREDENTIAL =
+            ElasticSearchConfiguration.Credential.of(DEFAULT_USERNAME, DEFAULT_PASSWORD);
+
+        private static final String NGINX_SERVICE = "nginx";
+        private static final int COMPILATION_BY_PASS_DUMMY_PORT = 0;
```

Comment:
???
why compilation ?
bypass or by passing ?

Reply:
IDK how to name it, just a dummy value to pass to the method. It does nothing